src/core/tools/kilocode/searchAndReplaceTool.ts

```diff
@@ -130,6 +128,12 @@ export async function searchAndReplaceTool(
 			return
 		}
 
+		const useCrLf = fileContent.includes("\r\n")
+		const validSearch =
+			useCrLf && !params.old_str.includes("\r\n") ? params.old_str.replace("\n", "\r\n") : params.old_str
```

Comment:
note that if the AI decides to mix \r\n and \n this replacement is not going to happen - I guess that's OK?